server/src/main/java/io/crate/planner/WhereClauseOptimizer.java

```diff
@@ -201,10 +207,10 @@ public static DetailedQuery optimize(EvaluatingNormalizer normalizer,
         }
 
         WhereClauseValidator.validate(query);
-        return new DetailedQuery(query, docKeys, partitionValues, clusteredBy);
+        return new DetailedQuery(query, docKeys, partitionValues, clusteredBy, queryHasPkSymbolsOnly(query, table));
```

Comment:
Could we move this call into the constructor?  It would ensures that its value and the `query` are consistent.  Right now it would be possible to pass values as parameters that violate this.